jdaviz/configs/mosviz/helper.py

```diff
@@ -429,15 +437,14 @@ def load_data(self, spectra_1d=None, spectra_2d=None, images=None,
         msg = ""
 
         if directory is not None and Path(directory).is_dir():
-            if instrument not in ('nirspec', 'niriss'):
-                msg = ("Ambiguous MOS Instrument: Only JWST NIRSpec and "
-                       "NIRISS folder parsing is fully supported; falling back to NIRSpec parsing.")
-                warnings.warn(msg)
-                instrument = "nirspec"
+            if instrument not in ('nirspec', 'niriss', 'nircam'):
+                msg = ("Ambiguous MOS Instrument: Only JWST NIRSpec, NIRCam, and "
+                       "NIRISS folder parsing are currently supported")
+                raise ValueError(msg)
             if instrument.lower() == "nirspec":
                 super().load_data(directory, parser_reference="mosviz-nirspec-directory-parser")
-            elif instrument.lower() == "niriss":
-                self.load_niriss_data(directory)
+            elif instrument.lower() in ("niriss", "nircam"):
```

Comment:
Actual check is already done above, so no need to repeat here.
```suggestion
            else:  # niriss or nircam
```